runtime/lua/vim/lsp/buf.lua

```diff
@@ -1405,11 +1405,16 @@ end
 
 --- Perform an incremental selection at the cursor position based on ranges given by the LSP. The
 --- `direction` parameter specifies the number of times to expand the selection. Negative values
---- will shrink the selection.
+--- will shrink the selection. The `use_sync` is optional. When true, the request is performed
+--- synchronously. Operator-pending mode requires synchronous execution.
 ---
 --- @param direction integer
-function M.selection_range(direction)
+--- @param use_sync? boolean
+function M.selection_range(direction, use_sync)
```

Comment:
I think instead of adding a `use_sync` flag this should probably follow the pattern from https://github.com/neovim/neovim/pull/34523 and add a callback to make it `vim.async` ready. Until `vim.async` is ready one can then use the callback to set a flag to block on with `vim.wait`.

Alternatively: Could make this in general synchronous. Not sure if there's much value for it to be async, given that opposed to something like completion it is unusual to continue editing while getting the selection range.